packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-has-one-relationship/BookUpdateForm.jsx

```diff
@@ -0,0 +1,394 @@
+  const { id, book, onSuccess, onError, onSubmit, onCancel, onValidate, onChange, overrides, ...rest } = props;
+  const initialValues = {
+    name: undefined,
+    primaryAuthor: undefined,
+  };
+  const [name, setName] = React.useState(initialValues.name);
+  const [primaryAuthor, setPrimaryAuthor] = React.useState(initialValues.primaryAuthor);
+
+  // grab records for suggestion
+  const authorRecords = useDataStoreBinding({
+    type: 'collection',
+    model: Author,
+  }).items;
+
+  // generated from valueMappings config
+  const getDisplayValue = {
+    primaryAuthor: (record) => record?.name,
+  };
+
+  const [errors, setErrors] = React.useState({});
+  const resetStateValues = () => {
+    setName(initialValues.name);
+    setPrimaryAuthor(primaryAuthor);
+    setCurrentPrimaryAuthorValue(undefined);
+    setCurrentPrimaryAuthorDisplayValue(undefined);
+  };
+
+  // fetch model data to fill form with preset values
+  const [bookRecord, setBookRecord] = React.useState(book);
+  React.useEffect(() => {
+    const queryData = async () => {
+      const record = id ? await DataStore.query(Book, id) : book;
+      setBookRecord(record);
+      // get existing relationship record
+      const existingAuthor = record ? await DataStore.query(Author, (a) => a.id.eq(record.authorId)) : undefined;
+      // set existing record on autocomplete if it exists
+      setPrimaryAuthor(existingAuthor);
+      setCurrentPrimaryAuthorValue(existingAuthor && existingAuthor[0].id);
+      setCurrentPrimaryAuthorDisplayValue(existingAuthor && existingAuthor[0]?.name);
+    };
+    queryData();
+  }, [id, book]);
+
+  React.useEffect(resetStateValues, [bookRecord]);
+
+  // displayValue (string) and value (Model) must be set separately
+  const [currentPrimaryAuthorValue, setCurrentPrimaryAuthorValue] = React.useState(undefined);
+
+  const [currentPrimaryAuthorDisplayValue, setCurrentPrimaryAuthorDisplayValue] = React.useState(undefined);
+  const primaryAuthorRef = React.createRef();
+
+  const validations = {
+    name: [],
+    primaryAuthor: [],
+  };
+
+  // new arg so we're validating string, not Model
+  const runValidationTasks = async (fieldName, currentValue, getDisplayValue) => {
+    const value = getDisplayValue ? getDisplayValue(currentValue) : currentValue;
+    let validationResponse = validateField(value, validations[fieldName]);
+    const customValidator = fetchByPath(onValidate, fieldName);
+    if (customValidator) {
+      validationResponse = await customValidator(value, validationResponse);
+    }
+    setErrors((errors) => ({ ...errors, [fieldName]: validationResponse }));
+    return validationResponse;
+  };
+  return (
+    <Grid
+      as="form"
+      rowGap="15px"
+      columnGap="15px"
+      padding="20px"
+      onSubmit={async (event) => {
+        event.preventDefault();
+        let modelFields = {
+          name,
+          primaryAuthor: primaryAuthor?.id, // pass in id field
```

Comment:
Wouldn't this be `authorId` according to the schema?

Reply:
Good catch, it should be `authorId: primaryAuthor?.id`